CharlieBackend.Core/Models/ResultModel/Result.cs

```diff
@@ -0,0 +1,75 @@
+﻿using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Threading.Tasks;
+
+namespace CharlieBackend.Core.Models.ResultModel
+{
+    public class Result<T>
+    {
+
+        public Task<T> TransferredData { get; private set; }
+
+        public Error Error { get; private set; }
+
+        public static Task<Result<T>> ReturnData(Task<T> transferredData)
+        {
+            if (transferredData == null)
```

Comment:
If someone passed null into this method, it means that something gone wrong. throw ArgumentNullException and mention about this in method doc
https://docs.microsoft.com/en-us/dotnet/csharp/codedoc#exception